command/pki_reissue_intermediate.go

```diff
@@ -0,0 +1,297 @@
+package command
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"net"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/posener/complete"
+)
+
+type PKIReIssueCACommand struct {
+	*BaseCommand
+
+	flagConfig          string
+	flagReturnIndicator string
+	flagDefaultDisabled bool
+	flagList            bool
+
+	flagKeyStorageSource string
+	flagNewIssuerName    string
+}
+
+func (c *PKIReIssueCACommand) Synopsis() string {
+	return "Uses a parent certificate and a template certificate to create a new issuer on a child mount"
+}
+
+func (c *PKIReIssueCACommand) Help() string {
+	helpText := `
+Usage: vault pki reissue PARENT TEMPLATE CHILD_MOUNT options
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (c *PKIReIssueCACommand) Flags() *FlagSets {
+	set := c.flagSet(FlagSetHTTP | FlagSetOutputFormat)
+	f := set.NewFlagSet("Command Options")
+
+	f.StringVar(&StringVar{
+		Name:       "type",
+		Target:     &c.flagKeyStorageSource,
+		Default:    "internal",
+		EnvVar:     "",
+		Usage:      `Options are “existing” - to use an existing key inside vault, “internal” - to generate a new key inside vault, or “kms” - to link to an external key.  Exported keys are not available through this API.`,
+		Completion: complete.PredictSet("internal", "existing", "kms"),
+	})
+
+	f.StringVar(&StringVar{
+		Name:    "issuer_name",
+		Target:  &c.flagNewIssuerName,
+		Default: "",
+		EnvVar:  "",
+		Usage:   `If present, the newly created issuer will be given this name`,
+	})
+
+	return set
+}
+
+func (c *PKIReIssueCACommand) Run(args []string) int {
+	// Parse Args
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	args = f.Args()
+
+	if len(args) < 3 {
+		c.UI.Error("Not enough arguments: expected parent issuer and child-mount location and some key_value argument")
+		return 1
+	}
+
+	stdin := (io.Reader)(os.Stdin)
+	userData, err := parseArgsData(stdin, args[3:])
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Failed to parse K=V data: %s", err))
+		return 1
+	}
+
+	// Check We Have a Client
+	client, err := c.Client()
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Failed to obtain client: %v", err))
+		return 1
+	}
+
+	parentIssuer := sanitizePath(args[0]) // /pki/issuer/default
+	intermediateMount := sanitizePath(args[2])
+
+	templateCertificateResp, err := client.Logical().Read(sanitizePath(args[1]))
```

Comment:
I'll fix this up across various commands, just be aware that it is possible that we get a nil, nil response here if the URL passed in wasn't a valid path. The list-intermediates has the same issue.